atomic_reactor/plugins/exit_koji_import.py

```diff
@@ -0,0 +1,262 @@
+
+    If metadata_only is set, the 'docker save' image will not be
+    uploaded, only the logs. The import will be marked as
+    metadata-only.
+
+    Runs as an exit plugin in order to capture logs from all other
+    plugins.
+    """
+
+    key = PLUGIN_KOJI_IMPORT_PLUGIN_KEY
+    is_allowed_to_fail = False
+
+    def __init__(self, tasker, workflow, kojihub, url,
+                 verify_ssl=True, use_auth=True,
+                 koji_ssl_certs=None, koji_proxy_user=None,
+                 koji_principal=None, koji_keytab=None,
+                 metadata_only=False, target=None,
+                 poll_interval=5):
+        """
+        constructor
+
+        :param tasker: DockerTasker instance
+        :param workflow: DockerBuildWorkflow instance
+        :param kojihub: string, koji hub (xmlrpc)
+        :param url: string, URL for OSv3 instance
+        :param verify_ssl: bool, verify OSv3 SSL certificate?
+        :param use_auth: bool, initiate authentication with OSv3?
+        :param koji_ssl_certs: str, path to 'cert', 'ca', 'serverca'
+        :param koji_proxy_user: str, user to log in as (requires hub config)
+        :param koji_principal: str, Kerberos principal (must specify keytab)
+        :param koji_keytab: str, keytab name (must specify principal)
+        :param metadata_only: bool, whether to omit the 'docker save' image
+        :param target: str, koji target
+        :param poll_interval: int, seconds between Koji task status requests
+        """
+        super(KojiImportPlugin, self).__init__(tasker, workflow)
+
+        self.kojihub = kojihub
+        self.koji_ssl_certs = koji_ssl_certs
+        self.koji_proxy_user = koji_proxy_user
+
+        self.koji_principal = koji_principal
+        self.koji_keytab = koji_keytab
+
+        self.metadata_only = metadata_only
+        self.target = target
+        self.poll_interval = poll_interval
+
+        self.namespace = get_build_json().get('metadata', {}).get('namespace', None)
+        osbs_conf = Configuration(conf_file=None, openshift_uri=url,
+                                  use_auth=use_auth, verify_ssl=verify_ssl,
+                                  namespace=self.namespace)
+        self.osbs = OSBS(osbs_conf, osbs_conf)
+        self.build_id = None
+
+    def get_output(self, metadatas):
+        """
+        Build the output entry of the metadata.
+
+        :return: list, containing dicts of partial metadata
+        """
+
+        for instance in metadatas['output']:
+            instance['buildroot_id'] = instance['arch'] + '-' + repr(instance['buildroot_id'])
+
+        return metadatas['output']
+
+    def get_buildroot(self, metadatas):
+        """
+        Build the buildroot entry of the metadata.
+
+        :return: list, containing dicts of partial metadata
+        """
+        for instance in metadatas['buildroots']:
+            instance['id'] = instance['container']['arch'] + '-' + repr(instance['id'])
```

Comment:
Maybe use `"{arch}-{id}".format(arch=instance['container']['arch'], id=instance['id'])`?

Might avoid ugliness with quotes if instance['id'] turned out to be a string of some sort, and issues with concatenation if one of these things ended up being/not being a unicode string.